test/test_nn.py

```diff
@@ -290,6 +306,57 @@ def _do_extra_tests(self, test_case, module, input, target):
                 _assertGradAndGradgradChecks(test_case, lambda x, y, z, *args, **kw: module(x, y, z),
                                              input + (target,) + params)
 
+    def test_cuda(self, test_case, dtype=None):
+        def convert_dtype(obj, dtype, requires_grad=False):
+            if isinstance(obj, Variable):
+                return Variable(obj.data.type(dtype), requires_grad=requires_grad)
+            elif torch.is_tensor(obj):
+                return obj.type(dtype)
```

Comment:
We get to simplify this when Variable and Tensor merge. CC @colesbury (should we wait till Tensor/Variable merge before merging this?)

Reply:
The PR is merged now. @ailzhang can you please update the code?